apps/mobile/src/components/amount-field/amount-field.utils.ts

```diff
@@ -0,0 +1,89 @@
+import { type CurrencyMode } from '@/features/send-refactor/types';
+
+import { currencyDecimalsMap } from '@leather.io/constants';
+import { type Currency, type MarketData, type Money } from '@leather.io/models';
+import { baseCurrencyAmountInQuote, createMoneyFromDecimal } from '@leather.io/utils';
+
+interface FormatPrimaryValueParams {
+  value: string;
+  currency: Currency;
+  showCurrency: boolean;
+  locale: string;
+}
+
+export function formatPrimaryValue({
+  value,
+  currency,
+  showCurrency,
+  locale,
+}: FormatPrimaryValueParams) {
+  const decimalSeparator = getDecimalSeparator(locale);
+  const decimalPart = value.split(decimalSeparator)[1];
+  const fractionDigits = decimalPart?.length ?? 0;
+
+  const formatter = new Intl.NumberFormat(locale, {
+    style: showCurrency ? 'currency' : 'decimal',
+    currency,
+    minimumFractionDigits: fractionDigits,
+    maximumFractionDigits: fractionDigits,
+  });
+
+  const formattedValue = formatter.format(Number(value));
+  // Ensure trailing decimal separator, as primary value is a live input,
+  return value.endsWith(decimalSeparator) ? formattedValue + decimalSeparator : formattedValue;
+}
+
+interface FormatSecondaryValueParams {
+  value: string;
+  currency: Currency;
+  locale: string;
+}
+
+export function formatSecondaryValue({ value, currency, locale }: FormatSecondaryValueParams) {
+  const formatter = new Intl.NumberFormat(locale, {
+    style: 'currency',
+    currency,
+    minimumFractionDigits: 0,
+    maximumFractionDigits: currencyDecimalsMap[currency],
+  });
+
+  return formatter.format(Number(value));
+}
+
+interface CalculateSecondaryValueParams {
+  value: string;
+  currency: Currency;
```

Comment:
isn't currency implicit based on base value of market data?

Reply:
It is, this is now outdated, thanks for pointing this out. I'm also realizing dynamic fiat currency support **has to be** plugged into the form within this PR (it's currently static) irrespective of Fiat input support that comes later. Working on this.